File: tm/NativeSampleModule.ts
/* eslint-disable prettier/prettier */
import { TurboModule, TurboModuleRegistry } from 'react-native';

export interface Spec extends TurboModule {
    readonly reverseString: (input: string) => string;
    readonly doubleNumber: (input: number) => number;
}

export default TurboModuleRegistry.getEnforcing<Spec>(
    'NativeSampleModule'
);

File: tm/NativeSampleModule.cpp
#include "NativeSampleModule.h"

namespace facebook::react {
NativeSampleModule::NativeSampleModule(std::shared_ptr<CallInvoker> jsInvoker)
    : NativeSampleModuleCxxSpec(std::move(jsInvoker)) {}

std::string NativeSampleModule::reverseString(jsi::Runtime &rt,
                                              std::string input) {
  return std::string(input.rbegin(), input.rend());
}

int NativeSampleModule::doubleNumber(jsi::Runtime &rt, int number) {
  return number * 2;
}
} // namespace facebook::react